code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.kt

```diff
@@ -0,0 +1,279 @@
+                    val offerType = format?.let { OfferType.from(it) }
+                        ?: OfferType.from(DataReader.getString(offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_TYPE))
+                    val language = DataReader.getStringList(offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_LANGUAGE)
+                    val characteristics = DataReader.getStringMap(offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_CHARACTERISTICS)
+
+                    val content = getContentFromOfferData(offerData) ?: DataReader.optString(
+                        offerData,
+                        OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_DELIVERYURL,
+                        null
+                    ) ?: run {
+                        Log.debug(
+                            OptimizeConstants.LOG_TAG,
+                            SELF_TAG,
+                            "Cannot create Offer object, provided data Map doesn't contain valid item data content or deliveryURL."
+                        )
+                        return null
+                    }
+
+                    return Builder(id, offerType, content)
+                        .setEtag(etag)
+                        .setScore(score)
+                        .setSchema(schema)
+                        .setMeta(meta)
+                        .setLanguage(language)
+                        .setCharacteristics(characteristics)
+                        .build()
+                } else {
+                    if (schema != OptimizeConstants.JsonValues.SCHEMA_TARGET_DEFAULT) {
+                        Log.debug(
+                            OptimizeConstants.LOG_TAG,
+                            SELF_TAG,
+                            "Cannot create Offer object, provided data Map doesn't contain valid item data."
+                        )
+                        return null
+                    }
+                    Log.trace(
+                        OptimizeConstants.LOG_TAG,
+                        SELF_TAG,
+                        "Received default content proposition item, Offer content will be set to an empty string."
+                    )
+                    return Builder(id, OfferType.UNKNOWN, "").build()
+                }
+            } catch (e: ClassCastException) {
+                Log.warning(OptimizeConstants.LOG_TAG, SELF_TAG, "Cannot create Offer object, provided data contains invalid fields.")
+                null
+            } catch (e: DataReaderException) {
+                Log.warning(OptimizeConstants.LOG_TAG, SELF_TAG, "Cannot create Offer object, provided data contains invalid fields.")
+                null
+            }
+        }
+
+        /**
+         * Creates a Map<String, Any> using this Offer's attributes.
+         *
+         * @return Map<String, Any> containing Offer data.
+         */
+        @JvmStatic
+        fun Offer.toEventData(): Map<String, Any> {
```

Comment:
The original method was not static. While it is not a public method and should be fine, could you please share why you decided to change the implementation?

Reply:
Also this needs to be marked internal since the original method was not public